src/main/java/com/cgm/infolab/controller/ChatApiMessagesController.java

```diff
@@ -46,6 +44,11 @@ public List<ChatMessageDto> getAllMessagesGeneral(@RequestParam(required = false
             log.info("Non sono stati trovati messaggi nella room specificata");
         }
 
+        chatService.setAllMessagesAsDownloadedInRoom(
```

Comment:
perchè non usi le entity dei messaggi che hai già in mano da `chatService.getAllMessagesGeneral` per sapere gli id dei messaggi di cui aggiornare il timestamp?

dovrebbe venire query più semplice